os/kernel/task/task_recover.c

```diff
@@ -114,6 +115,8 @@
 
 void task_recover(FAR struct tcb_s *tcb)
 {
+	/* This task is being deleted. Do PM post cleaning required for this thread */
+	pm_recover(tcb);
```

Comment:
Better to add a config check around it as its not required if pm or pm metrics are not enabled.

Reply:
Sorry, I missed the else part of the definition,

```
#define pm_recover(tcb)         (0)
```

Ignore this.

Reply:
Is there a reason to call this first?

if no, let's move after mq_recover

Reply:
I kept it first, to get all the tcb information. If we kept it last we will lose wdog and semaphore information.

Reply:
it would be better to move the wdog used in pm sleep to wd_recover and move location after mq_recover

Reply:
Done